mne_bids/write.py

```diff
@@ -1922,28 +1929,100 @@ def mark_bad_channels(ch_names, descriptions=None, *, bids_path,
+
+
+@verbose
+def mark_channels(ch_names, descriptions=None, *, bids_path,
+                  status='bad', overwrite=False, verbose=None):
+    """Update status and description of channels in an existing BIDS dataset.
+
+    Parameters
+    ----------
+    ch_names : str | list of str
+        The names of the channel(s) to mark as bad. Pass an empty list in
+        combination with ``overwrite=True`` to mark all channels as good.
+    descriptions : None | str | list of str
+        Descriptions of the reasons that lead to the exclusion of the
+        channel(s). If a list, it must match the length of ``ch_names``.
+        If ``None``, no descriptions are added.
+    bids_path : mne_bids.BIDSPath
+        The recording to update. The :class:`mne_bids.BIDSPath` instance passed
+        here **must** have the ``.root`` attribute set. The ``.datatype``
+        attribute **may** be set. If ``.datatype`` is not set and only one data
+        type (e.g., only EEG or MEG data) is present in the dataset, it will be
+        selected automatically.
+    status : str | list of str
+        The status of the channels ('good', or 'bad'). Default is 'bad'. If it
+        is a list, then must be a list of 'good', or 'bad' that has the same
+        length as ``ch_names``.
+    overwrite : bool
+        If ``False``, only update the information of the channels passed via
+        ``ch_names``, and leave the rest untouched. If ``True``, update the
+        information of **all** channels: mark the channels passed via
+        ``ch_names`` as bad, and all remaining channels as good, also
+        discarding their descriptions.
+    %(verbose)s
+
+    Examples
+    --------
+    Mark a single channel as bad.
+
+    >>> root = Path('./mne_bids/tests/data/tiny_bids').absolute()
+    >>> bids_path = BIDSPath(subject='01', task='rest', session='eeg',
+    ...                      datatype='eeg', root=root)
+    >>> mark_channels('C4', bids_path=bids_path, verbose=False)
+
+    Mark multiple channels as bad, and add a description as to why.
+
+    >>> bads = ['C3', 'PO10']
+    >>> descriptions = ['very noisy', 'continuously flat']
+    >>> mark_channels(bads, descriptions, bids_path=bids_path,
+    ...                   verbose=False)
+
+    Mark two channels as bad, and mark all others as good by setting
+    ``overwrite=True``.
+
+    >>> bads = ['C3', 'C4']
+    >>> mark_channels(bads, bids_path=bids_path,
+    ...                   overwrite=True, verbose=False)
+
+    Mark all channels as good by passing an empty list of bad channels, and
+    setting ``overwrite=True``.
+
+    >>> mark_channels([], bids_path=bids_path, overwrite=True,
+    ...                   verbose=False)
+
+    Mark all channels with a new description, while keeping them as a "good"
+    channel.
+
+    >>> descriptions = ['resected', 'resected']
+    >>> mark_channels(['C3', 'C4'], bids_path=bids_path,
+    ...               descriptions=descriptions, status='good')
```

Comment:
worth mentioning what files it updates? `channels.tsv` ?

Reply:
Added to doc string